client/my-sites/promote-post-i2/utils/index.ts

```diff
@@ -174,15 +174,20 @@ export const canCancelCampaign = ( status: string ) => {
 
 export const getPagedBlazeSearchData = (
 	mode: 'campaigns' | 'posts',
-	campaignsData?: PagedBlazeSearchResponse
+	pagedData?: PagedBlazeSearchResponse
 ): PagedBlazeContentData => {
-	const lastPage = campaignsData?.pages?.[ campaignsData?.pages?.length - 1 ];
+	const lastPage = pagedData?.pages?.[ pagedData?.pages?.length - 1 ];
 	if ( lastPage ) {
 		const { has_more_pages, total_items } = lastPage;
 
-		const foundContent = campaignsData?.pages
-			?.map( ( page: any ) => page[ mode ] )
-			?.flat() as Campaign[];
+		let foundContent = pagedData?.pages
+			?.map( ( item: any ) => item[ mode ] )
+			?.flat()
+			?.filter( ( item: any ) => 'undefined' !== typeof item );
+
+		if ( foundContent?.length && 'campaigns' === mode ) {
```

Comment:
> I think we can remove this part. The .flat() converts the result as any, so it won't give you a typescript error if you don't cast it to Campaign[].

I tried to do so, but then it produces an error:
<img width="986" alt="image" src="https://github.com/Automattic/wp-calypso/assets/115007291/2bc8780b-ee49-4146-b5ad-5a3bd3c773d2">

<img width="903" alt="image" src="https://github.com/Automattic/wp-calypso/assets/115007291/15193c6a-541b-42a2-b18c-6c9f984e794e">

Once I've returned `flat` call, it started working as usual.

> The approach we had before was wrong; the casting as Campaign[] was not correct, because that method was also used by the posts query.
> 
> Maybe is a good idea to create another ticket and review all of our types, maybe refactor them and move them to a type file, Right now we have some living in `calypso/data/promote-post/types` and others just inside the hooks.

That's definitely a great suggestion. Applied it in https://github.com/Automattic/wp-calypso/pull/78776/commits/e3062227a814129c7cfea59b8c200b355553c972.

Looking forward to the next round of your review! 🙌